### PR TITLE
Jc/add crud for support users

### PR DESCRIPTION
**Pull request by JamieCleare2525**
## Context

Support Users should be able to add new Users to Organisations (Providers/Schools)

## Changes proposed in this pull request

- Index page listing users in an Organisation
- Show page showing details of users
- Adding/Invitation process for Users onto a Organisation

## Guidance to review

(As a Placements Support User)
**Index**
- View an Organisation with Users
- A list of users should be displayed
screencapture-placements-localhost-3000-support-organisations-fffd6b99-f9ad-4e34-b8a3-9b3cc1fba66e-users-2024-01-16-17_13_35

**Show**
- View an Organisation with Users
- Click on the names of one of the Users
- A page showing the User's details should be displayed
screencapture-placements-localhost-3000-support-organisations-fffd6b99-f9ad-4e34-b8a3-9b3cc1fba66e-users-6877644e-8ce4-41b0-83eb-74008e68c9fb-2024-01-16-17_14_45

**New/Check/Create**
- View an Organisation
- Click "Invite user"
- Enter the User's details
- Click "Continue"
- The details you have entered should be displayed
- Click "Add user"
- The User should be added to the list of Users on that Organisation

## Link to Trello card

https://trello.com/c/gvZLRSCB/119-replicate-user-crud-for-support-placements

## Screenshots

https://github.com/DFE-Digital/itt-mentor-services/assets/17162488/5d3f6c0a-4fc0-49c1-a01e-c3462d240cdd

### CODE REVIEW

File: app/views/placements/support/organisations/users/_check_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        <%= f.hidden_field :first_name, value: @user.first_name %>
+        <%= f.hidden_field :last_name, value: @user.last_name %>
+        <%= f.hidden_field :email, value: @user.email %>
+
+        <label class="govuk-label govuk-label--l">
+          <span class="govuk-caption-l"><%= t(".caption", organisation_name: organisation.name) %></span>
+          <%= t(".title") %>
+        </label>
+
+        <dl class="govuk-summary-list">
```
Comment (elceebee) [inline app/views/placements/support/organisations/users/_check_form.html.erb:15]: Is there a reason we are not using the govuk_summary_list helper? I think it's a bit neater.

File: app/views/placements/support/organisations/users/_details.html.erb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<%# locals: (user: nil, organisation: nil) -%>
+<div class="govuk-width-container">
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+      <span class="govuk-caption-l"><%= organisation.name %></span>
+      <h2 class="govuk-heading-l"><%= user.full_name %></h2>
+
+      <dl class="govuk-summary-list">
```
Comment (elceebee) [inline app/views/placements/support/organisations/users/_details.html.erb:8]: same as above, if possible, could we use the summary_list helper?

File: app/views/placements/support/providers/users/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-full">
+      <h1 class="govuk-heading-l"><%= @provider.name %></h1>
+      <%= render SecondaryNavigationComponent.new do |component| %>
+        <% component.with_navigation_item t(".secondary_navigation.details"), "#" %>
+        <% component.with_navigation_item t(".secondary_navigation.users"), placements_support_provider_users_path(@provider) %>
+        <% component.with_navigation_item t(".secondary_navigation.mentors"), "#" %>
+        <% component.with_navigation_item t(".secondary_navigation.placements"), "#" %>
+        <% component.with_navigation_item t(".secondary_navigation.providers"), "#" %>
+      <% end %>
```
Comment (elceebee) [inline app/views/placements/support/providers/users/index.html.erb:13]: Depending on who merges in first, we can use the partial in my PR for this.

File: app/views/placements/support/providers/users/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    </div>
+
+    <div class="govuk-grid-column-two-thirds">
+      <h2 class="govuk-heading-m"><%= t(".heading") %></h2>
+      <%= govuk_button_to(t(".add_user"), new_placements_support_provider_user_path(@provider), method: :get) %>
+
+      <table class="govuk-table">
```
Comment (elceebee) [inline app/views/placements/support/providers/users/index.html.erb:20]: There is also a `govuk_table` helper, here is an example of it in use:

https://github.com/DFE-Digital/register-trainee-teachers/blob/bd71de101065fc2cd9825295ca8053b55b0d236e/app/views/guidance/dates_and_deadlines.html.erb#L13

File: spec/services/user_invite_service_spec.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
+require "rails_helper"
+
+RSpec.describe UserInviteService do
```
Comment (elceebee) [inline spec/services/user_invite_service_spec.rb:3]: 👍 Good to have a unit test for this!

**Review by elceebee — APPROVED** [review]
Just a couple of comments, but no blockers from my POV, so I've gone ahead and approved (though nice to have responses before you merge!)

And could you confirm that there is another ticket in the wings for adding the "Change" links to the "Check your answers" bit of the journey? It -- and the back button on that page -- should navigate back to the form with the data you've already inputted. It looks like it's out of scope for this PR, but I just want to make sure we don't lose track of it.

File: app/views/placements/support/providers/users/index.html.erb (hunk 4 above)
Comment (JamieCleare2525) [reply app/views/placements/support/providers/users/index.html.erb]: resolved in [853411c](https://github.com/DFE-Digital/itt-mentor-services/pull/125/commits/853411c3dc199496f425ee6adeb80fc5690ef163)

File: app/views/placements/support/organisations/users/_check_form.html.erb (hunk 1 above)
Comment (JamieCleare2525) [reply app/views/placements/support/organisations/users/_check_form.html.erb]: resolved in [853411c](https://github.com/DFE-Digital/itt-mentor-services/pull/125/commits/853411c3dc199496f425ee6adeb80fc5690ef163)

File: app/views/placements/support/organisations/users/_details.html.erb (hunk 2 above)
Comment (JamieCleare2525) [reply app/views/placements/support/organisations/users/_details.html.erb]: resolved in [853411c](https://github.com/DFE-Digital/itt-mentor-services/pull/125/commits/853411c3dc199496f425ee6adeb80fc5690ef163)

**Review by CatalinVoineag — APPROVED** [review]

**Pull request merged**